import-automation/executor/schedule_update_import.py

```diff
@@ -0,0 +1,234 @@
+    the branch master. Therefore, any local changes in the repo or import script
+    will not be reflected.
+
+    Args:
+        cfg: a configs.ExecutorConfig object with all the fields required for
+            the update script. Since it contains authentication information,
+            the configs should be read from a secure location, e.g. a Cloud
+            bucket, and then parsed and passed here as a configs.ExecutorConfig
+            object.
+        absolute_import_path: a string specifying the import's path in the
+            following format:
+                <path_to_directory_relative_to_repository_root>:<import_name>
+            example:
+                scripts/us_usda/quickstats:UsdaAgSurvey
+        import_script_args: a list of strings, each to be used as a command
+            line arg for the import script,
+            e.g. ['--flag1=value1', '--flag2=value2'].
+        local_repo_dir: the full path to the GitHub repository on local. The
+            path shoud be provided to the root  directory of the repo,
+            e.g. `<base_path_on_disk>/data`.
+
+    Returns:
+        An import_executor.ExecutionResult object.
+    """
+    # Update the configs with user script args, if provided.
+    if import_script_args:
+        cfg.user_script_args = import_script_args
+
+    executor = import_executor.ImportExecutor(
+        uploader=file_uploader.GCSFileUploader(
+            project_id=cfg.gcs_project_id,
+            bucket_name=cfg.storage_prod_bucket_name),
+        github=github_api.GitHubRepoAPI(
+            repo_owner_username=cfg.github_repo_owner_username,
+            repo_name=cfg.github_repo_name,
+            auth_username=cfg.github_auth_username,
+            auth_access_token=cfg.github_auth_access_token),
+        config=cfg,
+        local_repo_dir=local_repo_dir)
+
+    return executor.execute_imports_on_update(absolute_import_path)
+
+
+def main(_):
+    mode = _FLAGS.mode
+    absolute_import_path = _FLAGS.absolute_import_path
+    import_script_args = _FLAGS.import_script_args
+
+    if not _FLAGS.config_project_id:
+        raise Exception("Flag: config_project_if must be provided.")
+
+    if not mode or (mode not in ['update', 'schedule']):
+        raise Exception('Flag: mode must be set to \'update\' or \'schedule\'')
+
+    if not import_target.is_absolute_import_name(absolute_import_path):
+        raise Exception(
+            'Flag: absolute_import_path is invalid. Path should be like:'
+            'scripts/us_usda/quickstats:UsdaAgSurvey')
+
+    # Converting string to list
+    args_list = import_script_args.split(' ')
+    if type(args_list) != type([]):
+        raise Exception(
+            'Flag: import_script_args could not be parsed into a list.')
+
+    # Get the root repo directory (data). Assumption is that this script is being
+    # called from a path within the data repo.
+    cwd = os.getcwd()
+    repo_dir = cwd.split("data")[0] + "data"
```

Comment:
Would it be useful to have a way to run this script to execute from repo rather than local dir always?

Reply:
My thought is that there exist two ways to run it off of the latest master state of the repo:

1. You can run this same script from the master branch locally and it would be in the same state as running it off of the master state on the remote repo.
2. There is a workflow which allows these updates to be executed from Cloud Scheduler which will always download the latest repo and then execute.

The reason I wanted to keep this disabled on the local script is that our repo is ~500Mb and for every execution of this script to have to wait to download the repo first before executing is painful.